canonical/assets.ubuntu.com
# Update Python dependencies

## Done

- Update python dependencies
- Replace the python dependency pilbox with Pillow: [pilbox](https://github.com/agschwender/pilbox) uses Pillow under the hood, the problem is that pilbox isn't maintained anymore where it's using an old version of Pillow ~5.0.0 whereas the latest version is 9.5.0. Which makes it hard to install the requirements on Python +3.8 and forces the use of python 3.6 .

## QA

- Check out this feature branch
- Run the site using the command `dotrun`
- View the site locally in your web browser at: http://0.0.0.0:8017
- Run through the following [QA steps](https://canonical-web-and-design.github.io/practices/workflow/qa-steps.html)
- Make sure that creating asset is working
- Make sure that the assets list is working
- Make sure that an asset link is working

## QA the usage of Pillow

- Locally, Upload a JPG and PNG image to the asset server
- Here is a complete set of operation that uses the Pillow library: http://0.0.0.0:8017/v1/your-imagepng?op=region,rotate,resize&max-width=100&max-height=100&rect=0,0,500,500&deg=-50
- Make sure that this link works with your images and applies the proper transformations
- Make an operation with the wrong parameters (missing an option for `rect`):  http://0.0.0.0:8017/v1/your-imagepng?op=region,rotate,resize&max-width=100&max-height=100&rect=0,0,500&deg=-50
- Make sure that the error is shown with the error `400`
